app/src/main/java/com/eis/phoneringer/MainActivity.java

```diff
@@ -1,14 +1,127 @@
 package com.eis.phoneringer;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.os.Build;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
 
+import com.eis.phoneringer.structure.AppManager;
+import com.eis.phoneringer.structure.PasswordManager;
+import com.eis.phoneringer.structure.ReceivedMessageListener;
+import com.eis.phoneringer.structure.RingCommand;
+import com.eis.smslibrary.SMSHandler;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis.smslibrary.exceptions.InvalidSMSMessageException;
+import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+import com.eis.smslibrary.listeners.SMSSentListener;
+
+/**
+ * Test app: through this MainActivity we can test out our basic library for sending ring commands
+ *
+ * @author Alberto Ursino
+ */
 public class MainActivity extends AppCompatActivity {
 
+    private static Button ringButton = null;
+    private static EditText phoneNumber = null;
```

Comment:
why are these `static`?

Reply:
Because it seems correct to me since it makes no sense to instantiate the same one twice, for example, EditText.
Am I wrong? Ready to learn.

Reply:
I don't think those should be static either: the Activity is a normal instance and could have different views (those set on `setContentView(R.layout.*)`) therefore graphic elements shouldn't be static

Reply:
`ringButton` is only used in method `onCreate()`, so you could instantiate it there as a local variable, and `phoneNumber` could be left where it is, but I don't understand why you made it static. Does it need to be accessed and have the same value in different instances of MainActivity? Are we even going to have multiple instances of MainActivity needing to share that value? If not, then I would remove the `static` keyword. Android Studio always warns that placing Context classes in static fields is a memory leak, so it should be avoided unless absolutely necessary.

Reply:
I'll try your solution @giovannivelludo